coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
       }
 
       /**
-       * Override this method so that there will be explicit and correct table alias for selected fields, which is necessary for
-       * data type derivation on SqlNodes
+       * Override this method so that there will be explicit table alias for all the fields, which is necessary for
+       * data type derivation on struct fields.
+       *
+       * For the following input SQL:
+       * CREATE TABLE default.complex(s struct(name:string, age:int));
+       * SELECT split(complex.s.name, ' ') name_array
+       * FROM default.complex
+       * WHERE complex.s.age = 25;
+       *
+       * The input RelNode is:
+       * LogicalProject(name_array=[split($0.name, ' ')])
+       *   LogicalFilter(condition=[=($0.age, 25)])
+       *     LogicalTableScan(table=[[hive, default, complex]])
+       *
+       * With this override, the converted SqlNode is:
+       * SELECT `split`(`complex`.`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex` AS `complex`
+       * WHERE `complex`.`s`.`age` = 25
+       *
+       * Without this override, the converted SqlNode is:
+       * SELECT `split`(`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex`
+       * WHERE `s`.`age` = 25
+       *
+       * Without this override, if we want to get the data type of a struct field like `s`.`name`, validation will fail
+       * because Calcite uses {@link org.apache.calcite.rel.type.StructKind#FULLY_QUALIFIED} for standard SQL and it
+       * expects each struct field to be referenced explicitly with table alias in the method
```

Comment:
could you rephrase "each struct field" to emphasize "each field inside the struct"? 
Iirc, the struct field, `s`, is not an issue. the field inside the struct, `s.name` fails data type derivation unless fully qualified. Pl correct if this is not the case